Assets/Scripts/Entity/Enemy/EnemyDrop.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyDrop : MonoBehaviour
+{
+    /* Called when an enemy has died. Spawns a heart with random probability.
+    */
+    public static void DropHeart(Transform t, float healthDropProbability, float heartTypeProbability)
+    {
+        var heartType = GameObject.Find("Half-Heart");
+        var heartTypeChance = Random.Range(0f, 1f);
+        if(heartTypeChance >= heartTypeProbability)
+        {
+            heartType = GameObject.Find("Heart");
```

Comment:
`GameObject.Find` only returns if the gameobject is active within the scene (or any other active scene). Which the heart will not be in a normal scene....  Instead you must follow something like Alex's suggestion above to store a reference to a prefab(s).